src/Equations/Fluid/LCFSLossRateTerm.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * An implementation of a simple loss term modeling the loss of runaways when the flux surfaces are broken due to plasma drift.
+ * A negative and linear in n_RE runaway generation rate term using a, from the input script, provided timescale t_loss. The term
+ * is applied to all radial grid points outside of the last closed flux surface (LCFS). 
+ */
+
+#include "DREAM/DREAMException.hpp"
+#include "DREAM/Equations/Fluid/LCFSLossRateTerm.hpp"
+
+using namespace DREAM;
+
+/**
+ * Constructor.
+ */
+LCFSLossRateTerm::LCFSLossRateTerm(FVM::Grid *grid, FVM::UnknownQuantityHandler *unknowns, FVM::Grid *operandGrid, real_t sf, real_t* t_loss, len_t userGivenPsiEdge_t0, real_t PsiEdge_t0) : 
+    RunawaySourceTerm(grid, unknowns), FVM::DiagonalComplexTerm(grid, unknowns, operandGrid), unknowns(unknowns), scaleFactor(sf), t_loss(t_loss), 
+    userGivenPsiEdge_t0(userGivenPsiEdge_t0), psi_edge_t0(PsiEdge_t0), 
+    id_psi(unknowns->GetUnknownID(OptionConstants::UQTY_POL_FLUX)) {
+
+    SetName("LCFSLossRateTerm");
+    this->GridRebuilt();
+}
+
+/**
+ * Destructor
+ */
+LCFSLossRateTerm::~LCFSLossRateTerm(){
+    Deallocate();
+}           
+
+
+/**
+ * Called after the grid is rebuilt; (re)allocates memory
+ * for all quantities
+ */
+bool LCFSLossRateTerm::GridRebuilt(){
+
+    this->DiagonalComplexTerm::GridRebuilt();
+    Deallocate();
+    this->GammaLoss = new real_t[nr];
+    
+    return true;
+}
+
+
+/**
+* Set weights for term to be multiplied with unknown (n_RE). 
+*/
+void LCFSLossRateTerm::SetGammaLoss(){
+
+    for (len_t ir = 0; ir < nr; ir++){ // Empty GammaLoss
+        this->GammaLoss[ir] = 0;
+    }
+    
+    FindRadiusOfLCFS(); // Find ir_LCFS
+    
+    for (len_t ir = 0; ir < nr; ir++){
+        this->GammaLoss[ir] = -StepFunction(ir) / this->t_loss[ir]; // Change GammaLoss from 0 if ir>ir_LCFS
+    }
+}
+
+
+
+/**
+* Find current radial point of last closed flux surface
+* and find psi_edge_t0 if not provided in input
+*/
+void LCFSLossRateTerm::FindRadiusOfLCFS(){
+    
+    if(this->userGivenPsiEdge_t0 == 0) {
+        this->psi_edge_t0 = unknowns->GetUnknownInitialData(id_psi)[nr-1];
+    }
+    
+    // Find if the sign of PsiDiff() should be +1 or -1
+    if(this->signFixed == false) { 
+        real_t psi_mid_init = unknowns->GetUnknownInitialData(id_psi)[0]; 
+        real_t psi_edge_init = unknowns->GetUnknownInitialData(id_psi)[nr-1];
+        if(psi_mid_init-psi_edge_init > 0) { this->sign = 1; }
+        this->signFixed = true;
+    }
+    
+    bool exists = false;
```

Comment:
A quite picky comment, but wouldn't it be possible to write lines 82-90 in a shorter way like

```
for(len_t ir = 0; ir < nr; ir++)
    if(this->sign * PsiDiff(ir)<0)
        this->ir_LCFS=(int_t)ir-1;
```
?